adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala

```diff
@@ -89,20 +94,71 @@ private[rdd] object NucleotideContigFragmentRDD extends Serializable {
 
     NucleotideContigFragmentRDD(rdd, sd)
   }
+
+  def apply(rdd: RDD[NucleotideContigFragment],
+            sequences: SequenceDictionary): NucleotideContigFragmentRDD = {
+    RDDBoundNucleotideContigFragmentRDD(rdd, sequences)
+  }
+}
+
+case class ParquetUnboundNucleotideContigFragmentRDD private[rdd] (
```

Comment:
To the first two, sure, let me do those.

To the last one, I prefer putting the load code inside of the unbound RDD and making it lazy. IIRC, this way we avoid some of the overhead (e.g., calculating splits) until computation occurs.